Always show most recent blog posts in sidebar. Fixes #28.

jennifer-richards opened the pull request:
This always uses `now()` as the limiting time when selecting blogs to list in the sidebar.

jennifer-richards committed:
Always show most recent blog posts in sidebar. Fixes #28.

meadmaker commented:
Could you help me understand how this affects which blog posts are selected to show?  It looks more like it affects what date is shown for the blog post, but that the actual selection of the blog posts is unaffected.

jennifer-richards commented:
What's going on is the argument is using a [field lookup](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#field-lookups). That is, `d__lt=some_date` becomes `WHERE d < some_date` in the SQL query. (I.e., the equivalent of `:d.lt => some_date` with DM in Ruby)

The other piece is that the original code used `d__lt = self.coalesced_published_date() or datetime.now()`. The right-hand side was the published date if one was defined, otherwise the present time. I've just changed it to always use the present time as the limit.

Does that help?

meadmaker commented:
I must have been out of it yesterday, because I was completely misreading that bit of code (I misread the "if" as a "def", and thought it was a one-line function definition).  

Now that I see what the code is doing, I have another question - is the if/else construct still worthwhile, or would it be better to just have it always look up the last five items?

jennifer-richards commented:
That's a good question. It does not look to me like there's any reason for that if block now. (It appears to me that it was preventing the `datetime.now()` from ever being called in the first place.)

I will take it out unless @rjsparks knows why it's needed.

rjsparks commented:
The block can go away.

I worry, though, that we should look deeper. The block was written when the idea was "show the 5 blog entries before this one". Now we want "The 5 most recent blog entries". Or to be more specific, "The 5 most recent _published_ blog entries". Please explore what siblings means a little and see if there's a more efficient way to get that set, and that the filter we're using gets only published pages.

jennifer-richards commented:
Ah, glad I pinged you. Makes sense, will do.

jennifer-richards committed:
Remove unnecessary if statement

jennifer-richards committed:
Drop redundant siblings filter, add comments

The BlogPage.siblings() method only returns live pages - no need
to filter based on the current time.

jennifer-richards commented:
I had a look at this and it seems to me that there's no need to filter on date/time at all - the `BlogPage.siblings` method includes a filter to only return "live" pages and returns a most-recent-first ordered list of sibling pages.

I did a bit of testing with published / unpublished / scheduled publish pages and the behavior seems to be as you describe - the sidebar contains the most recent 5 published entries.
